From 5f6d0007854b4b8835ba945c4237f598317a0701 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 12 Apr 2020 22:46:17 +0300
Subject: [PATCH 131/133] add Amoled Black overlay

	new file:   packages/overlays/AmoledDarkUIOverlay/Android.mk
	new file:   packages/overlays/AmoledDarkUIOverlay/AndroidManifest.xml
	new file:   packages/overlays/AmoledDarkUIOverlay/res/values-night/colors.xml
	modified:   packages/overlays/Android.mk
---
 .../overlays/AmoledDarkUIOverlay/Android.mk   | 30 ++++++++++++++
 .../AmoledDarkUIOverlay/AndroidManifest.xml   | 24 +++++++++++
 .../res/values-night/colors.xml               | 41 +++++++++++++++++++
 packages/overlays/Android.mk                  |  1 +
 4 files changed, 96 insertions(+)
 create mode 100644 packages/overlays/AmoledDarkUIOverlay/Android.mk
 create mode 100644 packages/overlays/AmoledDarkUIOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/AmoledDarkUIOverlay/res/values-night/colors.xml

diff --git a/packages/overlays/AmoledDarkUIOverlay/Android.mk b/packages/overlays/AmoledDarkUIOverlay/Android.mk
new file mode 100644
index 00000000000..43da57693e8
--- /dev/null
+++ b/packages/overlays/AmoledDarkUIOverlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := AmoledDarkUI
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := AmoledDarkUIOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/AmoledDarkUIOverlay/AndroidManifest.xml b/packages/overlays/AmoledDarkUIOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..0dfafe8dd60
--- /dev/null
+++ b/packages/overlays/AmoledDarkUIOverlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.darkui.amoled"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="android" android:category="android.theme.customization.darkui" android:priority="1"/>
+    <application android:label="Amoled Black" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/AmoledDarkUIOverlay/res/values-night/colors.xml b/packages/overlays/AmoledDarkUIOverlay/res/values-night/colors.xml
new file mode 100644
index 00000000000..052fb3896b4
--- /dev/null
+++ b/packages/overlays/AmoledDarkUIOverlay/res/values-night/colors.xml
@@ -0,0 +1,41 @@
+<!--
+/**
+ * Copyright (c) 2020, CAOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <color name="notification_primary_text_color_dark">#ffe5e5e5</color>
+    <color name="notification_secondary_text_color_dark">#b3e5e5e5</color>
+    <color name="notification_default_color_dark">#ffe5e5e5</color>
+    <color name="notification_material_background_color">@android:color/black</color> <!-- material_grey_900 -->
+    <color name="primary_material_dark">@android:color/black</color> <!-- material_grey_900 -->
+    <color name="primary_material_settings">@android:color/black</color> <!-- material_grey_900 -->
+    <color name="primary_dark_material_dark">@android:color/black</color>
+    <color name="primary_dark_material_settings">@android:color/black</color> <!-- material_grey_850 -->
+    <color name="primary_dark_device_default_dark">@android:color/black</color> <!-- material_grey_850 -->
+    <color name="primary_dark_device_default_settings">@android:color/black</color> <!-- main settings bg -->
+    <color name="background_floating_device_default_dark">@android:color/black</color>
+    <color name="secondary_material_settings">#ff0f1012</color> <!-- material_grey_800 -->
+    <color name="tertiary_material_settings">#ff151618</color> <!-- material_grey_600 -->
+    <color name="quaternary_material_settings">#ff202126</color> <!-- material_grey_200 -->
+    <color name="material_grey_900">@android:color/black</color>
+    <color name="material_grey_850">@android:color/black</color>
+    <color name="material_grey_800">#ff0f1012</color>
+    <color name="material_grey_600">#ff151618</color>
+    <color name="material_grey_300">#ff1a1b1e</color>
+    <color name="material_grey_200">#ff202126</color>
+    <color name="material_grey_100">#ff3c4043</color>
+    <color name="material_grey_50">#ffdadce0</color>
+</resources>
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index 15f99ae1e74..de19cbe7138 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -81,6 +81,7 @@ LOCAL_REQUIRED_MODULES := \
 	QSPanel34Overlay \
 	QSPanel43Overlay \
 	QSPanel55Overlay \
+	AmoledDarkUIOverlay \
 	CharcoalDarkUIOverlay \
 	GrayDimDarkUIOverlay \
 	MidnightDarkUIOverlay \
-- 
2.17.1

